kernel/h1/src/hil/flash/hardware.rs

```diff
@@ -14,8 +14,19 @@
 
 use kernel::ReturnCode;
 
-/// The interface between the flash driver and the (real or fake) flash module.
+/// The bank to perform an operation on.
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub enum Bank {
+    Unknown,
+    Zero,
+    One,
```

Comment:
Why not just:

```rust
pub enum Bank {
    Zero,
    One,
}
```

then use `Option<Bank>` (or a similarly-structured custom enum type) wherever you need a third value.

Reply:
In particular, this would make `H1bHw::trigger` simpler.

Reply:
I tried to compromise wrt. required changes. But you're right, removing `Unknown` is absolutely better. Please see latest changes.